lighthouse-core/test/lib/network-request-test.js

```diff
@@ -0,0 +1,160 @@
+
+      assert.equal(req.transferSize, 100);
+      req._updateTransferSizeForLightrider();
+      assert.equal(req.transferSize, 100);
+    });
+  });
+
+  describe('update fetch stats for Lightrider', function() {
+    function getRequest() {
+      const req = new NetworkRequest();
+      // units = seconds
+      req.startTime = 0;
+      req.endTime = 2;
+      req.responseReceivedTime = 1;
+      req.timing = {
+        requestTime: -1,
+        proxyStart: -1,
+        proxyEnd: -1,
+        dnsStart: -1,
+        dnsEnd: -1,
+        connectStart: -1,
+        connectEnd: -1,
+        sslStart: -1,
+        sslEnd: -1,
+        workerStart: -1,
+        workerReady: -1,
+        sendStart: -1,
+        sendEnd: -1,
+        pushStart: -1,
+        pushEnd: -1,
+        receiveHeadersEnd: -1,
+      };
+
+      // units = ms
+      req.responseHeaders = [{name: NetworkRequest.HEADER_TOTAL, value: 10000},
+        {name: NetworkRequest.HEADER_TCP, value: 5000},
+        {name: NetworkRequest.HEADER_REQ, value: 2500},
+        {name: NetworkRequest.HEADER_SSL, value: 1000},
+        {name: NetworkRequest.HEADER_RES, value: 2500}];
+      return req;
+    }
+
+    it('does nothing if not Lightrider', function() {
+      const req = getRequest();
+      global.isLightrider = false;
+
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 2);
+      assert.equal(req.responseReceivedTime, 1);
+      req._updateTimingsForLightrider();
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 2);
+      assert.equal(req.responseReceivedTime, 1);
+    });
+
+    it('does nothing if no TotalTime', function() {
+      const req = getRequest();
+      global.isLightrider = true;
+      req.responseHeaders = req.responseHeaders.filter(item => item.name
+        !== NetworkRequest.HEADER_TOTAL);
+
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 2);
+      assert.equal(req.responseReceivedTime, 1);
+      req._updateTimingsForLightrider();
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 2);
+      assert.equal(req.responseReceivedTime, 1);
+    });
+
+    it('does nothing if Header timings don\'t add up', function() {
+      const req = getRequest();
+      global.isLightrider = true;
+      req.responseHeaders = [{name: NetworkRequest.HEADER_TOTAL, value: 10000},
+        {name: NetworkRequest.HEADER_TCP, value: 5001},
+        {name: NetworkRequest.HEADER_REQ, value: 2500},
+        {name: NetworkRequest.HEADER_SSL, value: 1000},
+        {name: NetworkRequest.HEADER_RES, value: 2500}];
+
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 2);
+      assert.equal(req.responseReceivedTime, 1);
+      req._updateTimingsForLightrider();
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 2);
+      assert.equal(req.responseReceivedTime, 1);
+    });
+
+    it('does nothing if timing is not initialized', function() {
```

Comment:
seems like this test doesn't do anything since the headers aren't set?

Reply:
Right, it is confirming that the header timing isn't changed if no header is set.  Like if we made a change that always messed this up and set the value no matter what the header said it would break this b/c we shouldn't be messing with the timing w/o a header.